5.0 Draft/IDD/IDDs Service Registry/eu.arrowhead.service-discovery-http-json.yml

```diff
@@ -0,0 +1,631 @@
+    for consumers to register, deregister and search for _service records_, provided by themselves
+    or by other service providers.
+    
+    Each service record represents a currently provided service and contains all details necessary
+    to _contact_ that service. Such details will always include network addresses and a _service_
+    _type identifier_, apart from other optional information. The service type identifier is a
+    reference to the particular message protocol implemented by the provided service. Such a
+    protocol typically builds upon a base protocol, such as HTTP or MQTT, and includes endpoints,
+    topics or other operators, as well as message types specific to the provided service. For the
+    consumption of the service referred to by a service record to be possible, the consumer in
+    question must already implement the protocol referred to by its service type identifier.
+
+   
+    ## Compression and Language
+
+    An implementation of this service interface _may_ be designed to support compression and/or
+    human-readable error texts in different languages than English, as described in 
+    [RFC 9110, Section 12](https://www.rfc-editor.org/rfc/rfc9110#name-content-negotiation). Not
+    using compression _must_ be supported and be the default. Also, the default language for error
+    messages _must_ be American English (`en-US`).
+
+    ## Managing Record Life-Cycles
+
+    In a typical use case, this service interface is meant to hold records for all service providers
+    currently part of its context. When a service provider joins or leaves a context, it is expected
+    to register and deregister its services, respectively. However, service providers can fail or be
+    detached from the context before they get a chance to deregister their services. This can make
+    it relevant to monitor service providers and temporarily suppress or permanently prune records
+    associated with unhealthy or inaccessible providers. Implementations of this service interface
+    are free to suppress and/or prune records in whatever way best supports the use cases they are
+    designed for.
+
+    ## Record Ownership
+
+    It is the provider indicated inside each record that is the owner of the record, and not the
+    consumer registering it. This means that a consumer may be forbidden from registering or
+    deregistering a record for certain or all other service providers, even if it may register
+    and/or deregister records with itself as provider.
+
+    ## Provider Identifiers
+
+    Every service instance registered via this service interface _must_ be associated with a unique
+    identifier. In contexts where authentication is not used, providers _must_ chose their own
+    identifiers. Those identifiers _should_ remain consistent between restarts of those providers.
+    In contexts where authentication is used, the identifier _must_ be derived as follows: TODO.
```

Comment:
Not sure if this is relevant given the recent discussions we have had. Since authentication is done on a system level rather than service level then we do not have to propagate that to the service instance identifiers. It might make sense to do so practically from an operational perspective, but the framework should not in my opinion enforce this.

Reply:
So, how would you propose that this is formulated?